contracts/refs/CoreRefV2.sol

```diff
@@ -0,0 +1,241 @@
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFourRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFiveRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4,
+        bytes32 role5
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender) ||
+                _core.hasRole(role5, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    /// @notice set pausable methods to paused
+    function pause() public override onlyGuardianOrGovernor {
+        _pause();
+    }
+
+    /// @notice set pausable methods to unpaused
+    function unpause() public override onlyGuardianOrGovernor {
+        _unpause();
+    }
+
+    /// @notice address of the Core contract referenced
+    /// @return ICore implementation address
+    function core() public view override returns (ICoreV2) {
+        return _core;
+    }
+
+    /// @notice address of the Volt contract referenced by Core
+    /// @return IVolt implementation address
+    function volt() public view override returns (IVolt) {
+        return _core.volt();
+    }
+
+    /// @notice address of the Vcon contract referenced by Core
+    /// @return IERC20 implementation address
+    function vcon() public view override returns (IERC20) {
+        return _core.vcon();
+    }
+
+    /// @notice volt balance of contract
+    /// @return volt amount held
+    function voltBalance() public view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice vcon balance of contract
+    /// @return vcon amount held
+    function vconBalance() public view override returns (uint256) {
+        return vcon().balanceOf(address(this));
+    }
+
+    /// @notice set new reference to core
+    /// only callable by governor
+    /// @param newCore to reference
+    function setCore(address newCore) external onlyGovernor {
+        address oldCore = address(_core);
+        _core = ICoreV2(newCore);
+
+        emit CoreUpdate(oldCore, newCore);
+    }
+
+    /// @notice sweep target token, this shouldn't be needed, however it is a backup
+    /// in case a contract holds tokens and isn't a PCV Deposit
+    /// @param token to sweep
+    /// @param to recipient
+    /// @param amount of token to be sent
+    function sweep(
```

Comment:
This is a pretty large expansion of the scope of governance - funds could be swept from most contracts. Kind of concerning to me that this is `onlyPCVController` and you have some automated contracts that can move funds with that role - why not make it `onlyGovernor`?